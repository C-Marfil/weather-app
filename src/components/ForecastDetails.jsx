import React from "react";
import PropTypes from "prop-types";
import WindDirectionIcon from "./WindDirectionIcon";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__temperature" data-testid="ancestor">
        <div data-testid="descendant">Max Temp: {temperature.max}&deg;C</div>
        <div>Min Temp: {temperature.min}&deg;C</div>
      </div>
      <div className="forecast-details__wind">
        Wind: {wind.speed}mph
        <WindDirectionIcon direction={wind.direction} />
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
    </div>
  );
};

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
  }).isRequired,
};

export default ForecastDetails;
