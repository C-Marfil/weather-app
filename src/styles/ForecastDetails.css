.forecast-details {
    background-color: rgb(70, 56, 56);
    color: #fff;
    opacity: 0.8;
    border-radius: 10px;
    padding: 10px;
    display: flex;
    flex-direction: column;
    font-size: 25px;
    margin-right: 33vw;
    margin-left: 34vw;
    margin-top: 5vh;
}

.Max::before {
    content:"⬆️ ";
}

.Min::before {
    content:"⬇️ ";
}