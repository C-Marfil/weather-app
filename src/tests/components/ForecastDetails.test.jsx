import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    date: 2222222,
    humidity: 80,
    wind: {
      speed: 30,
      direction: "s",
    },
    temperature: {
      max: 33,
      min: 22,
    },
  };

  it("Renders Forecast Details", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct element", () => {
    const { getByTestId } = render(<ForecastDetails forecast={validProps} />);
    const ancestor = getByTestId("ancestor");
    const descendant = getByTestId("descendant");

    expect(ancestor).toContainElement(descendant);
  });
});
