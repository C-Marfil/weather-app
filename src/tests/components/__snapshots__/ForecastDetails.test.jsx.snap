// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details Renders Forecast Details 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
      data-testid="ancestor"
    >
      <div
        data-testid="descendant"
      >
        Max Temp: 33°C
      </div>
      <div>
        Min Temp: 22°C
      </div>
    </div>
    <div
      class="forecast-details__wind"
    >
      Wind: 30mph s
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 80%
    </div>
  </div>
</DocumentFragment>
`;
